test/test_transforms_tensor.py

```diff
@@ -728,6 +585,132 @@ def test_center_crop(device):
+    # test_ten_crop
+    ("ten_crop", "TenCrop", 10)
+])
+@pytest.mark.parametrize('fn_kwargs', [
+    {"size": (5,)},
+    {"size": [5, ]},
+    {"size": (4, 5)},
+    {"size": [4, 5]}
+])
+def test_x_crop(func, method, out_length, fn_kwargs, device):
+    meth_kwargs = fn_kwargs
+
+    fn = getattr(F, func)
+    scripted_fn = torch.jit.script(fn)
+
+    tensor, pil_img = _create_data(height=20, width=20, device=device)
+    transformed_t_list = fn(tensor, **fn_kwargs)
+    transformed_p_list = fn(pil_img, **fn_kwargs)
+    assert len(transformed_t_list) == len(transformed_p_list)
+    assert len(transformed_t_list) == out_length
+    for transformed_tensor, transformed_pil_img in zip(transformed_t_list, transformed_p_list):
+        _assert_equal_tensor_to_pil(transformed_tensor, transformed_pil_img)
+
+    transformed_t_list_script = scripted_fn(tensor.detach().clone(), **fn_kwargs)
+    assert len(transformed_t_list) == len(transformed_t_list_script)
+    assert len(transformed_t_list_script) == out_length
+    for transformed_tensor, transformed_tensor_script in zip(transformed_t_list, transformed_t_list_script):
+        assert_equal(transformed_tensor, transformed_tensor_script)
+
+    # test for class interface
+    fn = getattr(T, method)(**meth_kwargs)
+    scripted_fn = torch.jit.script(fn)
+    output = scripted_fn(tensor)
+    assert len(output) == len(transformed_t_list_script)
+
+    # test on batch of tensors
+    batch_tensors = _create_data_batch(height=23, width=34, channels=3, num_samples=4, device=device)
+    torch.manual_seed(12)
+    transformed_batch_list = fn(batch_tensors)
+
+    for i in range(len(batch_tensors)):
+        img_tensor = batch_tensors[i, ...]
+        torch.manual_seed(12)
+        transformed_img_list = fn(img_tensor)
+        for transformed_img, transformed_batch in zip(transformed_img_list, transformed_batch_list):
+            assert_equal(transformed_img, transformed_batch[i, ...])
+
+
+@cpu_only
+@pytest.mark.parametrize('method', ["FiveCrop", "TenCrop"])
+def test_x_crop_save(method):
+    fn = getattr(T, method)(size=[5, ])
+    scripted_fn = torch.jit.script(fn)
+    with get_tmp_dir() as tmp_dir:
+        scripted_fn.save(os.path.join(tmp_dir, "t_op_list_{}.pt".format(method)))
+
+
+@cpu_only
+@pytest.mark.parametrize('size', [32, 34, 35, 36, 38])
+def test_resize_int(size):
+    # TODO: Minimal check for bug-fix, improve this later
+    x = torch.rand(3, 32, 46)
+    t = T.Resize(size=size)
+    y = t(x)
+    # If size is an int, smaller edge of the image will be matched to this number.
+    # i.e, if height > width, then image will be rescaled to (size * height / width, size).
+    assert isinstance(y, torch.Tensor)
+    assert y.shape[1] == size
+    assert y.shape[2] == int(size * 46 / 32)
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('dt', [None, torch.float32, torch.float64])
+@pytest.mark.parametrize('size', [[32, ], [32, 32], (32, 32), [34, 35]])
```

Comment:
@NicolasHug @datumbox I have removed the int sizes from here and hence also removed the if/else later. To make sure we are still testing int for Resize, we have test_resize_int.